clang/lib/Sema/SemaTemplateDeductionGuide.cpp

```diff
@@ -1186,17 +1281,34 @@ void DeclareImplicitDeductionGuidesForTypeAlias(
         NewParam->setScopeInfo(0, I);
         FPTL.setParam(I, NewParam);
       }
-      auto *Transformed = cast<FunctionDecl>(buildDeductionGuide(
-          SemaRef, AliasTemplate, /*TemplateParams=*/nullptr,
+
+      QualType ReturnType =
+          cast<FunctionProtoType>(FunctionType->getType())->getReturnType();
+      if (!FunctionType)
+        continue;
```

Comment:
Is this check really necessary? Since you already performed a cast above, this can't be null.
Can you add a test case for this?

Reply:
Thanks for catching this! It's not necessary, I believe it's an erroneous leftover from a prior refactor. In the current code path that function type is always defined. I've removed the check accordingly.